server/form-pages/apply/risk-and-need-factors/further-considerations/additionalCircumstances.ts

```diff
@@ -0,0 +1,63 @@
+import type { TaskListErrors, YesOrNoWithDetail } from '@approved-premises/ui'
+import type { ApprovedPremisesApplication as Application } from '@approved-premises/api'
+import { Page } from '../../../utils/decorators'
+
+import TasklistPage from '../../../tasklistPage'
+import { yesOrNoResponseWithDetail } from '../../../utils'
+import { shouldShowContingencyPlanPages } from '../../../../utils/applications/shouldShowContingencyPlanPages'
+
+export const questionKeys = ['additionalCircumstances'] as const
+
+type QuestionKeys = (typeof questionKeys)[number]
+
+@Page({
+  name: 'additional-circumstances',
+  bodyProperties: ['additionalCircumstances', 'additionalCircumstancesDetail'],
+})
+export default class AdditionalCircumstances implements TasklistPage {
+  title = 'Additional circumstances'
+
+  questionPredicates = {
+    additionalCircumstances: 'do well in the past',
+  }
+
+  questions = {
+    additionalCircumstances: `Are there are any additional circumstances that have helped ${this.application.person.name} ${this.questionPredicates.additionalCircumstances}?`,
+  }
+
+  constructor(
+    public body: Partial<YesOrNoWithDetail<'additionalCircumstances'>>,
+    private readonly application: Application,
+  ) {}
+
+  previous() {
+    return 'arson'
+  }
+
+  next() {
+    return shouldShowContingencyPlanPages(this.application) ? 'contingency-plan-partners' : ''
+  }
+
+  response() {
+    return {
+      [this.questions.additionalCircumstances]: yesOrNoResponseWithDetail<QuestionKeys>(
+        'additionalCircumstances',
+        this.body,
+      ),
+    }
+  }
+
+  errors() {
+    const errors: TaskListErrors<this> = {}
+
+    if (!this.body.additionalCircumstances) {
+      errors.additionalCircumstances = `You must specify if ${this.application.person.name} there are any additional circumstances that have helped ${this.application.person.name} ${this.questionPredicates.additionalCircumstances}`
```

Comment:
Would probably replace the last `${this.application.person.name}` with `them`, as it reads a bit better.

Reply:
Good spot